### PR TITLE
Build business logic changes for release-branched helm support

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*

*Description of changes:*
Business logic changes necessary to support building release-branched helm charts.

Required for merging https://github.com/aws/eks-anywhere-build-tooling/pull/1287.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jaxesn — APPROVED** [review]

**Comment by ivyostosh** [comment]
Looks good to me. I will let Terry do one more review as he's more familiar with all the helm_require logics and if it impacts existing projects in some way I don't foresee.

File: Common.mk [hunk 1]
```diff
@@ -163,6 +163,8 @@ HELM_SOURCE_REPOSITORY?=$(REPO)
 HELM_SOURCE_IMAGE_REPO?=$(IMAGE_REPO)
 HELM_GIT_TAG?=$(GIT_TAG)
 HELM_TAG?=$(GIT_TAG)-$(GIT_HASH)
+HELM_USE_UPSTREAM_IMAGE?=false
+# HELM_DIRECTORY must be a relative path from project root to the directory that contains a `helm/` folder
```
[the remark this answers is not in the document]
Comment (jonathanmeier5) [reply Common.mk:167]: I think I misunderstood what exactly we were using it for, but I've double checked that the autoscaler chart still works regardless. I'll amend my single commit with your revision and re-push.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by jaxesn — APPROVED** [review]

**Comment by jonathanmeier5** [comment]
Merging with @jaxesn and @TerryHowe 's approval. Not clear to me why bot is still complaining.

**Pull request merged**